dallinger/prolific.py

```diff
@@ -149,23 +206,78 @@ def draft_study(
     ) -> dict:
         """Create a draft Study on Prolific, and return its properties."""
 
+        config = get_config()
+        config.load()
+
+        project_name = config.get("prolific_project")
+        workspace_name = config.get("prolific_workspace")
+        use_prolific_default_project = True
+
+        try:
+            # Get the workspace ID.  If it's not in Prolific, the function will raise an exception and create the workspace.
+            try:
+                workspace_id = self._translate_workspace_name(workspace_name)
+                use_prolific_default_project = False
+
+            except ProlificServiceNoSuchWorkspace:
+                # If the workspace does not exist in the '/workspaces/' endpoint, the '/studies/' endpoint will be
+                # called without a project key.
+
+                # Create a new workspace if it doesn't exist
+                response = self._req(
+                    method="POST",
+                    endpoint="/workspaces/",
+                    json={"title": workspace_name},
+                )
+                workspace_id = response["id"]
+                use_prolific_default_project = False
+
+        except Exception as e:
+            raise RuntimeError(f"Error finding or creating specified workspace: {e}")
```

Comment:
This is not the best way to reraise errors, because the original traceback might not be preserved. Try `raise from` (see e.g. https://stackoverflow.com/a/29442282)

Reply:
Ah, ok! Added a `from e` for both cases.